lib/checkleakautovar.cpp

```diff
@@ -331,6 +331,13 @@ void CheckLeakAutoVar::checkScope(const Token * const startToken,
         if (Token::Match(tok, "const %type%"))
             tok = tok->tokAt(2);
 
+        while (tok->str() == "(")
+            tok = tok->next();
+        while (tok && tok->isUnaryOp("*") && tok->astOperand1() && tok->astOperand1()->isUnaryOp("&"))
```

Comment:
I think we can assume that there is a token after a "(".

If `isUnaryOp` returns true then there is a `astOperand1`. So as far as I see the loop can safely be written like this:

    while (tok->isUnaryOp("*") && tok->astOperand1()->isUnaryOp("&"))
        tok = tok->astOperand1()->astOperand1();

And it shouldn't be necessary to check if tok is null below the loop.